src/application/selector/xml/XmlSelectorAdapter.ts

```diff
@@ -0,0 +1,34 @@
+import { DOMParser, XMLSerializer } from 'xmldom';
+import * as xpath from 'xpath';
+
+import DataSelectorPort from '../../../domain/common/port/DataSelectorPort';
+import XmlResultConverter from './XmlResultConverter';
+
+class XmlSelectorAdapter implements DataSelectorPort {
+  private readonly domParser = new DOMParser();
+  private readonly xmlSerializer = new XMLSerializer();
+
+  canHandle(contentType: string): boolean {
+    return ['xml', 'html'].includes(contentType);
+  }
+
+  select(data: string, path: string): string {
+    if (!path) {
+      return data;
+    }
+
+    const doc = this.domParser.parseFromString(data);
+    this.checkXmlValidity(doc);
+
+    const selected = xpath.select(path, doc);
+
+    return XmlResultConverter.toString(selected);
+  }
+
+  private checkXmlValidity(doc): void {
```

Comment:
We use a naming convention of `checkSomething()` with no return to indicate that this function validates its input and throws exception in case of any errors.

The logic is based on this specification, see `SyntaxError`:
https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer/serializeToString#Exceptions